wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_main-nav.scss

```diff
@@ -121,10 +121,19 @@ $footer-submenu: $submenu-color;
     .account {
         display: none;
     }
+
+    .submenu-trigger:after {
+        width: auto;
+        transition: transform .3s ease;
+        content: 'q';
+    }
 }
 
 .nav-submenu {
+    transition: max-height .2s;
     background: $submenu-color;
+    overflow: hidden;
+    max-height: 0;
```

Comment:
Please remove submenu animation from this PR. As you've said, there are trade-offs in animating the `max-height` so we should treat this as a separate change (that will need discussion).